gateway/src/http_entrypoints.rs

```diff
@@ -86,8 +69,8 @@ impl GatewayApi for GatewayImpl {
     /// Get info on an SP
     async fn sp_get(
         rqctx: RequestContext<Self::Context>,
-        path: Path<PathSp>,
-    ) -> Result<HttpResponseOk<SpState>, HttpError> {
+        path: Path<latest::component::PathSp>,
+    ) -> Result<HttpResponseOk<latest::component::SpState>, HttpError> {
```

Comment:
This is purely a style choice, but I think I would have imported all of the `latest` types so that this remained

```rust
    async fn sp_get(
        rqctx: RequestContext<Self::Context>,
        path: Path<PathSp>,
    ) -> Result<HttpResponseOk<SpState>, HttpError> {
```

and maybe would have done the same even in the `*-api` crate? And then only use explicit (non-`latest`) versions on the endpoints that need them.

I think this is mostly because adding `latest::submodule::` to every type referenced in this file causes a lot of pretty long lines and makes this diff noisier than it would be otherwise. It seems fine to me for an unadorned type to implicitly be "the latest". But I don't feel as strongly about this as I did other things we've already discussed.